app/controllers/api/service_templates_controller.rb

```diff
@@ -20,6 +20,14 @@ def create_resource(_type, _id, data)
 
     def edit_resource(type, id, data)
       catalog_item = resource_search(id, type, collection_class(:service_templates))
+      if data.key?("schedule_time")
+        schedule_time = data&.delete("schedule_time")
+        if schedule_time.nil?
+          catalog_item.delete_schedule
+        else
+          catalog_item.update_schedule(schedule_time)
+        end
+      end
       catalog_item.update_catalog_item(data.deep_symbolize_keys, User.current_user.userid)
```

Comment:
This line has the user as a parameter which implies rbac is being done, but your code is bypassing the rbac.  Maybe your code should go into the this method too?

Reply:
RBAC would have been done before this. Looking at the code, it seems like the user being saved in the workflow.